File: app/src/main/java/com/botijasoftware/utils/Mesh.java
package com.botijasoftware.utils;

import android.opengl.GLES20;
import com.botijasoftware.utils.materials.Material;

public class Mesh {
	
	public Mesh(String name, Texture texture, VertexBuffer vb, IndexBuffer ib) {
		mName = name;
		mVertexBuffer = vb;
		mIndexBuffer = ib;
		mTexture = texture;
	}

	public void LoadContent(ResourceManager resources) {}

	public void Draw() {
		GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, mTexture.getID());
		mVertexBuffer.Draw(mIndexBuffer);
	}
	
		
	public VertexBuffer mVertexBuffer;
	public IndexBuffer mIndexBuffer;	
	public Texture mTexture;
	public String mName;
	public Material mMaterial = null;
}

File: app/src/main/java/com/botijasoftware/utils/Mesh.kt
package com.botijasoftware.utils

import android.opengl.GLES20
import com.botijasoftware.utils.materials.Material

class Mesh(var mName: String, var mTexture: Texture, var mVertexBuffer: VertexBuffer, var mIndexBuffer: IndexBuffer) {

    fun LoadContent(resources: ResourceManager) {}

    fun Draw() {
        GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, mTexture.id)
        mVertexBuffer.Draw(mIndexBuffer)
    }

    var mMaterial: Material? = null
}
